students/aminavi/session04/trigram2.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3 
+import random
+
+def trigrams(x,n,m):
+    trigram_dic = {}
+    for i in range(len(x)-n): #looking to see how many words do we have in the sentence print(i)
+        pair = tuple(words[i:i + n]) #creating my keys looking at each word + 2 print(pair)
+        follower = words[i + n] #looking for each world after the combination
+        trigram_dic.setdefault(pair,[]).append(follower) # creating list a
+    #trigrams = list(random.choice(list(a.keys())))
+    words_count = m #how long the sentence going to be 
+    result = [] 
+    start_tuple = random.choice(list(trigram_dic.keys()))
+    result.append(start_tuple[0])
+    result.append(start_tuple[1])
+    while len(result) < words_count:
+        if start_tuple not in trigram_dic:
+            # result[-1] = result[-1]+"."
+            start_tuple = random.choice(list(trigram_dic.keys()))
+            result.append(start_tuple[0])
+            result.append(start_tuple[1])
```

Comment:
Seemingly correct logic overall, but not sure if this heuristic (of continuing, when the pair is not in the trigram dict) is going to be good. No need to change, though, I think.

Also, please try to modularize your code into multiple functions, as suggested in the problem description, not putting everything in one function like this. I see that the code doesn't seem too much, but it's always a good practice to modularize different responsibilities into different functions/classes/modules.